src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -104,7 +106,11 @@ public SwerveModule(int drivePort, int steerPort, double offsetRads) {
 
             steerAbsoluteEncoder = sparkMax.getAnalog(SparkMaxAnalogSensor.Mode.kAbsolute);
             steerAbsoluteEncoder.setPositionConversionFactor(STEER_VOLTS_TO_RADIANS);
-    
+
+            steerRelativeEncoder = sparkMax.getEncoder();
+            steerRelativeEncoder.setPositionConversionFactor(STEER_ROTATIONS_TO_RADIANS);
+            steerRelativeEncoder.setPosition(steerAbsoluteEncoder.getPosition()); // Set initial position to absolute value
```

Comment:
the position of the steer motor will not be limited to [0, 2pi]... will simply calling `steerPidController.setReference` be able to handle this?

Reply:
Not sure, will need to test. Ideally, the PID wrapping will be able to handle this? But if it doesn't ig we can copy over the old optimization-with-wraparound code and conditionally apply it based on the feedback sensor being used.

Reply:
i think we will need this. can we put it in, especially bc we won't really have time to test and we have a known working solution? we can leave the wrapping stuff commented in case it works, perhaps